modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/resolver/ValueResolverFactory.java

```diff
@@ -71,7 +75,7 @@ public <T> ValueResolver<T> of(String parameterName, MetadataType expectedType,
 
     if (isExpression(value)) {
       final String expression = (String) value;
-      resolver = getExpressionBasedValueResolver(expectedType, expression, modelProperties, expectedClass);
+      resolver = getExpressionBasedValueResolver(expectedType, expression, modelProperties, acceptsReferences, expectedClass);
```

Comment:
Exactly, initially I was considering making the change for any parameter that would accept references and support expressions, but then we decided to just make it for the config-refs to avoid messing with too many things at once. 
So, if you check inside `getExpressionBasedValueResolver` you'll see I also added a condition to make it only valid for `ValueResolvers` with `ConfigurationProvider` as `expectedClass`.
```
} else if (acceptsReferences && expectedClass.equals(ConfigurationProvider.class)) {
```
With this in mind, I don't know if the `acceptsReferences` check is redundant now, because config-ref parameters [always accept references](https://github.com/mulesoft/mule-extensions-api/blob/03d928d87c88a46e02a089df2db5bf0ab7fe4bb2/mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ConfigRefDeclarationEnricher.java#L111). But I thought it wouldn't hurt either and might protect us from issues down the line if we ever decide to have some other parameter that would accept a `ConfigurationProvider` but not necessarily a reference. For example: would it be possible to write a Java SDK operation that takes a parameter (not config-ref) with `ConfigurationProvider` as expected class, which supports expressions but does not accept references?

Reply:
I think it is not possible to declare such parameter, so it is fine 👌